zeppelin-zengine/src/main/java/org/apache/zeppelin/storage/LocalConfigStorage.java

```diff
@@ -95,17 +99,24 @@ public String loadCredentials() throws IOException {
   @Override
   public void saveCredentials(String credentials) throws IOException {
     LOGGER.info("Save Credentials to file: " + credentialPath);
-    writeToFile(credentials, credentialPath);
+    atomicWriteToFile(credentials, credentialPath);
   }
 
   private String readFromFile(File file) throws IOException {
     return IOUtils.toString(new FileInputStream(file));
   }
 
-  private void writeToFile(String content, File file) throws IOException {
-    FileOutputStream out = new FileOutputStream(file);
+  private void atomicWriteToFile(String content, File file) throws IOException {
+    File directory = file.getParentFile();
+    String fileNameParts[] = file.getName().split("\\.");
+    File tempFile = File.createTempFile(fileNameParts[0], fileNameParts[1], directory);
```

Comment:
when does the temp file get removed, if the write fails?

Reply:
Fixed. Please see comments in body of PR.